endpoint.cpp

```diff
@@ -346,35 +349,50 @@ void Endpoint::log_aggregate(unsigned int interval_sec)
     }
 }
 
-int UartEndpoint::open(const char *path, speed_t baudrate)
+UartEndpoint::~UartEndpoint()
+{
+    if (_change_baud_timeout) {
+        Mainloop::get_instance().del_timeout(_change_baud_timeout);
+        _change_baud_timeout = nullptr;
+    }
+}
+
+int UartEndpoint::open(const char *path, unsigned long baudrates[], size_t baud_size)
```

Comment:
instead of doing this, do the following:

* rename baudrate to "default_baudrate" - this is the baud we configure if the user never calls the method below
* add a set_speed() method